srv/isc-agent/plugins/tcp/http/models.py

```diff
@@ -0,0 +1,191 @@
+    FileResp = Column(BLOB)
+    CowrieRef = Column(Text)
+
+    def __repr__(self):
+        return f'''FileResp(ID={self.ID!r}, SigID={self.SigID!r}, FileNamePost={self.FileNamePost!r},
+        FileDataPost={self.FileDataPost!r}, FileTextPost={self.FileTextPost!r}, OSPath={self.OSPath!r},
+        FileResp={self.FileResp!r}, CowrieRef={self.CowrieRef!r})'''
+
+
+class Postlogs(Base):
+    __tablename__ = 'postlogs'
+
+    ID = Column(Integer, primary_key=True)
+    data = Column(Text)
+    headers = Column(Text)
+    address = Column(Text)
+    cmd = Column(Text)
+    path = Column(Text)
+    useragent = Column(Text)
+    vers = Column(Text)
+    formkey = Column(Text)
+    formvalue = Column(Text)
+    summary = Column(Text)
+
+    def __repr__(self):
+        return f'''Postlogs(ID={self.ID!r}, data={self.data!r}, headers={self.headers!r}, address={self.address!r}
+        cmd={self.cmd!r}, path={self.path!r}, useragent={self.useragent!r}, vers={self.vers!r}, 
+        formkey={self.formkey!r}, fomrvalue={self.formvalue!r}, summary={self.summary!r})'''
+
+
+class Files(Base):
+    __tablename__ = 'files'
+
+    ID = Column(Integer, primary_key=True)
+    RID = Column(Integer)
+    filename = Column(Text)
+    DATA = Column(BLOB)
+
+    def __repr__(self):
+        return f"Files(ID={self.ID!r}, RID={self.RID!r}, filename={self.filename!r}, DATA={self.DATA!r})"
+
+
+class Requests(Base):
+    __tablename__ = 'requests'
+
+    data = Column(Text, primary_key=True)
+    headers = Column(Text)
+    address = Column(Text)
+    cmd = Column(Text)
+    path = Column(Text)
+    useragent = Column(Text)
+    vers = Column(Text)
+    summery = Column(Text)
+    target = Column(Text)
+
+    def __repr__(self):
+        return f'''Requests(data={self.data!r}, headers={self.headers!r}, address={self.address!r}, 
+        cmd={self.cmd!r}, path={self.path!r}, useragent={self.useragent!r}, vers={self.vers!r},
+        summary={self.summery!r}, target={self.target!r})'''
+
+
+class Useragents(Base):
+    __tablename__ = 'useragents'
+
+    ID = Column(Integer, primary_key=True)
+    refid = Column(Integer)
+    useragent = Column(Text, unique=True)
+
+    def __repr__(self):
+        return f"Useragents(ID={self.ID!r}, refid={self.refid!r}, useragent={self.useragent!r})"
+
+
+class Responses(Base):
+    __tablename__ = 'responses'
+
+    ID = Column(Integer, primary_key=True)
+    RID = Column(Integer)
+    HeaderField = Column(Text)
+    dataField = Column(Text)
+
+    def __repr__(self):
+        return f'''Responses(ID={self.ID!r}, RID={self.RID!r}, 
+        HeaderField={self.HeaderField!r}, dataField={self.dataField!r})'''
+
+
+def build_models():
+    # create temporary engine in memory
+    # Establish connectivity with SQLalchemy engine
+    engine = create_engine("sqlite+pysqlite:///:memory:", echo=True, future=True)
+
+    # Front facing interface around python dictonary that stores table objects
+    metadata_obj = MetaData()
```

Comment:
What exactly does this do?

Reply:
"engine" defines the location of the database.  ":memory: "  is the location of this database. 

[Metadata](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.MetaData) this is the collection of Table objects in a Python dictionary with all the table details. Basically take the table data converts it and submits the proper SQL calls to the database.

Reply:
So Metadata is no longer needed since the registry does basically the same thing.